src/components/task-app/task-app.component.js

```diff
@@ -18,13 +18,75 @@ const TaskApp = () => {
     setTaskList((prevState) => [...prevState, task]);
   };
 
+  const taskCompletionToggle = (checkbox) => {
+    const taskId = checkbox.target.parentElement.attributes.id;
+
+    setTaskList((prevState) =>
+      prevState.map((task) => {
+        if (task.id !== taskId) {
+          return task;
+        } else {
+          return {
+            ...task,
+            completed: !task.completed,
+          };
+        }
+      })
+    );
+  };
+
+  const completeAllTasks = () => {
+    if (taskList.find((task) => !task.completed)) {
+      setTaskList((prevState) =>
+        prevState.map((task) => {
+          return {
+            ...task,
+            completed: true,
+          };
+        })
+      );
+    } else {
+      setTaskList((prevState) =>
+        prevState.map((task) => {
+          return {
+            ...task,
+            completed: false,
+          };
+        })
+      );
+    }
+  };
+
+  const handleTaskCompleted = (event) => {
+    const taskId = +event.target.parentElement.attributes.id.value;
+    const selectedTask = taskList.find((task) => task.id === taskId);
+
+    console.log(taskId);
+    console.log(selectedTask);
+
+    setTaskList((prevState) =>
+      prevState.map((task) => {
+        if (task.id === selectedTask.id) {
+          return { ...task, completed: !task.completed };
+        } else {
+          return task;
+        }
+      })
+    );
+  };
+
   return (
     <>
+      <button onClick={completeAllTasks}>Complete All</button>
       <button onClick={openForm}>Add Task</button>
       {formVisibility && (
         <TaskForm addNewTask={addNewTask} closeFormHandler={closeForm} />
       )}
-      <TaskList tasks={taskList} />
+      <TaskList
+        tasks={taskList}
+        taskCompletionToggle={taskCompletionToggle}
+        handleTaskCompleted={handleTaskCompleted}
```

Comment:
do `taskCompletionToggle` and `handleTaskCompleted` do the same thing?

Reply:
15095a96df1e6647cd8ad963dfb164ce921cdfe4 Removed the duplicated function.